### PR TITLE
Strong scaling data parallel acceleration for tensorflow.compat.v1

### PR DESCRIPTION
**Pull request by pescap**
Hi! At least it works. Some improvements can be done I guess.

### CODE REVIEW

**Comment by pescap** [comment]
`heat.py` tested with weak and strong scaling

File: deepxde/data/pde.py [hunk 1]
```diff
@@ -190,6 +185,18 @@ def train_next_batch(self, batch_size=None):
...
             config.comm.Bcast(self.train_x_bc, root=0)
         self.train_x = self.train_x_bc
+        if config.parallel_scaling == "strong":
+            # Split the training points over each rank.
+            # We drop last points in order to have the same number of points per rank
+            train_x_all = self.train_x_all
+            train_x_all_shape = list(
+                self.train_x_all.shape
+            )  # We transform to list to support item assignment
+            num_split = train_x_all_shape[0] // config.world_size
```
Comment (lululxvi) [inline deepxde/data/pde.py:195]: What if `train_x_all_shape[0] // config.world_size == 0`?

Comment (pescap) [reply deepxde/data/pde.py:195]: In this case, all the points are split. For example, if `world_size=2`, `num_domain=10` and `num_boundary=10`, I obtain at `pde line 199`:

```
print(train_x_all_shape[0], num_split, train_x_all.shape[0])
```
as 
```
10 10 20
```

Comment (pescap) [reply deepxde/data/pde.py:194]: Hi, let me know if something is not correct.

Comment (lululxvi) [reply deepxde/data/pde.py:194]: Yes, if the number of points < world_size, then there is no need to split, i.e., all ranks have all points.

Comment (pescap) [reply deepxde/data/pde.py:194]: I raise an error in this case, as the simulation would be pointless with only one point per rank.

File: deepxde/data/pde.py [hunk 2]
```diff
@@ -190,6 +185,23 @@ def train_next_batch(self, batch_size=None):
...
             config.comm.Bcast(self.train_x_bc, root=0)
         self.train_x = self.train_x_bc
+        if config.parallel_scaling == "strong":
+            # Split the training points over each rank.
+            # We drop last points in order to have the same number of points per rank
+            train_x_all_shape = list(
+                self.train_x_all.shape
+            )  # We transform to list to support item assignment
+            num_split = train_x_all_shape[0] // config.world_size
+            if num_split < config.world_size:
```
Comment (lululxvi) [inline deepxde/data/pde.py:195]: `if train_x_all_shape[0] < config.world_size`

Comment (pescap) [reply deepxde/data/pde.py:195]: sorry

File: deepxde/data/pde.py [hunk 3]
```diff
@@ -190,6 +185,23 @@ def train_next_batch(self, batch_size=None):
                 self.train_x_bc = np.empty(x_bc_shape, dtype=self.train_x_bc.dtype)
             config.comm.Bcast(self.train_x_bc, root=0)
         self.train_x = self.train_x_bc
+        if config.parallel_scaling == "strong":
+            # Split the training points over each rank.
+            # We drop last points in order to have the same number of points per rank
+            train_x_all_shape = list(
```
Comment (lululxvi) [inline deepxde/data/pde.py:191]:
```
            if len(self.train_x_all) < config.world_size:
                raise ValueError(
                    "The number of training points is smaller than the number of processes. Please use more points."
                )
            train_x_all_shape = list(
               ...
```

**Pull request merged**